components/dashboard/src/onboarding/StepUserInfo.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { User } from "@gitpod/gitpod-protocol";
+import { FC, useCallback, useState } from "react";
+import { TextInputField } from "../components/forms/TextInputField";
+import { useUpdateCurrentUserMutation } from "../data/current-user/update-mutation";
+import { useOnBlurError } from "../hooks/use-onblur-error";
+import { OnboardingStep } from "./OnboardingStep";
+
+type Props = {
+    user: User;
+    onComplete(user: User): void;
+};
+export const StepUserInfo: FC<Props> = ({ user, onComplete }) => {
+    const updateUser = useUpdateCurrentUserMutation();
+    // attempt to split provided name for default input values
+    const { first, last } = getInitialNameParts(user);
+
+    const [firstName, setFirstName] = useState(first);
+    const [lastName, setLastName] = useState(last);
+    const [emailAddress, setEmailAddress] = useState(User.getPrimaryEmail(user) ?? "");
+
+    const handleSubmit = useCallback(async () => {
+        const additionalData = user.additionalData || {};
+        const profile = additionalData.profile || {};
+
+        const updates = {
+            // we only split these out currently for form collection, but combine in the db
+            fullName: `${firstName} ${lastName}`,
+            additionalData: {
+                ...additionalData,
+                profile: {
+                    ...profile,
+                    emailAddress,
+                    lastUpdatedDetailsNudge: new Date().toISOString(),
+                },
+            },
+        };
+
+        try {
+            const updatedUser = await updateUser.mutateAsync(updates);
+            onComplete(updatedUser);
+        } catch (e) {
+            console.error(e);
+        }
+    }, [emailAddress, firstName, lastName, onComplete, updateUser, user.additionalData]);
+
+    const firstNameError = useOnBlurError("Please enter a value", !!firstName);
+    const lastNameError = useOnBlurError("Please enter a value", !!lastName);
```

Comment:
nit: not everyone has a last name, we should take that into account [[1](https://deedpolloffice.com/change-name/law)] - examples: [someone I know](https://twitter.com/DasSurma), [a list](https://en.wikipedia.org/wiki/List_of_legally_mononymous_people).

Although there may never be someone without a last name to see this flow, what would they put here if they don't have it? 😄

Reply:
Also, let's leave it to a follow-up.

Reply:
Yup, I'm on the same page as you. I anticipate this will need some changes. Fortunately we're not moving to store a distinct first/last name in the db, just a full name still.

Reply:
We could possibly skip the name split with the designs in https://github.com/gitpod-io/gitpod/pull/16501#pullrequestreview-1311386471. Cc @phimae